arangod/Cluster/ClusterCollectionMethods.cpp

```diff
@@ -769,3 +807,137 @@ ClusterCollectionMethods::createCollectionsOnCoordinator(
+  auto const& collectionID = std::to_string(col.id().id());
+  auto const& server = vocbase.server();
+  auto& ci = server.getFeature<ClusterFeature>().clusterInfo();
+  auto& agencyCache = server.getFeature<ClusterFeature>().agencyCache();
+
+  if (vocbase.replicationVersion() == replication::Version::TWO) {
+    VPackBufferUInt8 data;
+    VPackBuilder builder(data);
+    auto envelope = arangodb::agency::envelope::into_builder(builder);
+    auto properties = col.getCollectionProperties();
+    // NOTE: We could do this better with partial updates.
+    // e.g. we do not need to update the group if only the schema of the
+    // collection is modified
+
+    replication2::agency::CollectionGroupTargetSpecification::Attributes::
+        MutableAttributes groupUpdate;
+    {
+      groupUpdate.writeConcern = col.writeConcern();
+      groupUpdate.replicationFactor = col.replicationFactor();
+      groupUpdate.waitForSync = col.waitForSync();
+    }
+
+    auto groupBase = pathCollectionGroupInTarget(databaseName)
+                         ->group(std::to_string(col.groupID().id()));
+    auto groupPath = groupBase->str();
+    auto groupMutablesPath = groupBase->attributes()->mutables()->str();
+    auto colBase =
+        pathCollectionInTarget(databaseName)->collection(collectionID);
+    auto colPath = colBase->str();
+
+    auto writes = std::move(envelope).write();
+    // Update Mutable Properties of the collection
+    {
+      writes = std::move(writes).emplace_object(
+          colBase->schema()->str(),
+          [&](VPackBuilder& builder) { col.schemaToVelocyPack(builder); });
+
+      writes = std::move(writes).emplace_object(
+          colBase->computedValues()->str(), [&](VPackBuilder& builder) {
+            col.computedValuesToVelocyPack(builder);
+          });
+    }
+
+    // Update Mutable Properties of the group
+    writes = std::move(writes).emplace_object(
+        groupMutablesPath, [&](VPackBuilder& builder) {
+          velocypack::serialize(builder, groupUpdate);
+        });
+    // Increment the Group Version
+    writes = std::move(writes).inc(groupBase->version()->str());
+
+    // Preconditions: Database exists, Collection exists. Group exists.
+    auto preconditions = std::move(writes).precs();
+    preconditions = std::move(preconditions)
+                        .isNotEmpty(pathDatabaseInTarget(databaseName)->str());
+    preconditions = std::move(preconditions).isNotEmpty(colPath);
+    preconditions = std::move(preconditions).isNotEmpty(groupPath);
+    std::move(preconditions).end().done();
+
+    // Now data contains the transaction;
+    using namespace std::chrono_literals;
+    auto future = aac.sendWriteTransaction(120s, std::move(data))
+                      .thenValue(::reactToPreconditions)
+                      .thenValue(waitForOperationRoundtrip(ci));
+    return future.get();
+
+  } else {
+    AgencyPrecondition databaseExists("Plan/Databases/" + databaseName,
```

Comment:
Yes I think it actually is.
If you update the properties and drop the collection in parallel, the update could resurrect the dropped collection.
I have a hard time seeing this happening in reality in production though.
Will add a precondition that Collection still exists.

Reply:
Had to be a bit more drastic and added a precondition that the Collection was not CHANGED.
Otherwise we would race with property changes but more importantly with Index changes.